src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java

```diff
@@ -75,6 +93,71 @@ private static String detectMimeType(File file) {
 		return mimeType;
 	}
 
+	/**
+	 * Checks if passed-in file is a valid animated sticker: it first checks
+	 * if the file is a {@code gzip} archive, then it reads its content and verifies if it's a valid JSON.
+	 * Once JSON information are retrieved, they are validated against Telegram's requirements.
+	 *
+	 * @param file the file to check
+	 * @param mimeType the MIME type of the file
+	 * @return {@code true} if the file is compliant
+	 */
+	private static boolean isAnimatedStickerCompliant(File file, String mimeType) throws TelegramApiException {
+		if ("application/gzip".equals(mimeType)) {
+			String zipContent = "";
+
+			try (var gzipInputStream = new GZIPInputStream(new FileInputStream(file));
+				 var scanner = new Scanner(new InputStreamReader(gzipInputStream, StandardCharsets.UTF_8))) {
```

Comment:
JDK 9 introduces `InputStream#readAllBytes()` which can be used to create a new String using the constructor `String(byte[], Charset)`. This way we can avoid creating a scanner.

Reply:
Great suggestion, I just implemented it ✔️